spring-boot-actuator/src/main/java/org/springframework/boot/actuate/endpoint/mvc/SessionMvcEndpoint.java

```diff
@@ -0,0 +1,156 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.endpoint.mvc;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.actuate.endpoint.Endpoint;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.session.ExpiringSession;
+import org.springframework.session.FindByIndexNameSessionRepository;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurerAdapter;
+
+/**
+ * {@link MvcEndpoint} to expose actuator session.
+ *
+ * @author Eddú Meléndez
+ * @since 1.4.0
+ */
+@ConfigurationProperties("endpoints.session")
+public class SessionMvcEndpoint extends WebMvcConfigurerAdapter
+		implements MvcEndpoint {
+
+	/**
+	 * Endpoint URL path.
+	 */
+	private String path = "/session";
+
+	/**
+	 * Enable the endpoint.
+	 */
+	private boolean enabled = true;
+
+	@Autowired
+	private FindByIndexNameSessionRepository<? extends ExpiringSession> sessionRepository;
+
+	@RequestMapping(method = RequestMethod.GET)
+	public List<Session> result(@RequestParam String username) {
+		List<Session> sessions = new ArrayList<Session>();
+		for (ExpiringSession session : this.sessionRepository.findByIndexNameAndIndexValue(
+				FindByIndexNameSessionRepository.PRINCIPAL_NAME_INDEX_NAME, username)
+				.values()) {
+			sessions.add(new Session(session));
+		}
+		return sessions;
+	}
+
+	@RequestMapping(path = "/{sessionId}", method = RequestMethod.DELETE)
+	public void delete(@PathVariable String sessionId) {
+		ExpiringSession session = this.sessionRepository.getSession(sessionId);
+		if (session != null) {
+			this.sessionRepository.delete(sessionId);
+		}
+	}
+
+	public void setPath(String path) {
+		this.path = path;
+	}
+
+	@Override
+	public String getPath() {
+		return this.path;
+	}
+
+	public void setEnabled(boolean enabled) {
+		this.enabled = enabled;
+	}
+
+	public boolean isEnabled() {
+		return this.enabled;
+	}
+
+	@Override
+	public boolean isSensitive() {
+		return true;
+	}
+
+	@Override
+	public Class<? extends Endpoint> getEndpointType() {
+		return Endpoint.class;
+	}
+
+	/**
+	 * Session properties.
+	 */
+	public static class Session {
```

Comment:
IMO the representation of session should be modeled with more care considering the needs of potential consumer:

- ```creationTime``` and ```lastAccessedTime``` should be converted to ```Date```
- is ```maxInactiveIntervalInSeconds``` really needed here?
- consider providing at least _some_ info about session attributes